apps/dapp/frontend/components/ai/projection-confidence-band.tsx

```
@@ -0,0 +1,247 @@
+  )
+
+  if (points.length < 2) {
+    return (
+      <div className="rounded-xl border border-border bg-secondary/20 p-4 text-xs text-muted-foreground">
+        Not enough data points to render a confidence band.
+      </div>
+    )
+  }
+
+  // Estimated probability of reaching goal (last point's upper band vs goal).
+  const goalProbability = useMemo(() => {
```

ESLint

[error] React Hook "useMemo" is called conditionally. React Hooks must be called in the exact same order in every component render.

(react-hooks/rules-of-hooks)